### PR TITLE
Extend JS tutorial with Notifications, Plugins and Intents chapters

### PR DESCRIPTION
**Pull request by AleksTanev**
Extension of Glue Core JS tutorial, includes:
- Added json files in rest-server/data and extended functionality in server.js.
- Added Plugins, Notifications and Intents Chapters - updated the guide, solution and start folders.
- Workspaces is now Chapter 9 (previously 8) - changed 9.2 to take advantage of the Plugins API.
- Other small changes to Workspaces Chapter made (added/removed variables, changed the guide's text slightly etc.) to better sync with code from the new chapters.

### CODE REVIEW

**Review by flashd2n — CHANGES_REQUESTED** [review]
1. Extract the plugin into a separate file
2. Keep the `includeInWorkspaces` explanation
3. Rework the intents so that the intent is registered by a dedicated app - also make it export, instead of download

**Comment by AleksTanev** [comment]
I'm on it

File: tutorials/javascript/start/Clients/plugins/applicationsPlugin.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const fetchAppDefinitions = async (url) => {
```
Comment (flashd2n) [inline tutorials/javascript/start/Clients/plugins/applicationsPlugin.js:1]: should be created by the user

File: tutorials/javascript/start/Clients/plugins/layoutsPlugin.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const fetchWorkspaceLayoutDefinition = async (url) => {
+  const layoutsDefinitionsResponse = await fetch(url);
+  const layoutsDefinitions = await layoutsDefinitionsResponse.json();
+
+  return layoutsDefinitions;
+};
+
+const setupLayouts = async (glue, { url }) => {
```
Comment (flashd2n) [inline tutorials/javascript/start/Clients/plugins/layoutsPlugin.js:8]: should be created by the user

File: tutorials/javascript/start/client-details/index.html [hunk 3]
```diff
@@ -28,6 +28,7 @@ <h1 class="text-center">Client Details</h1>
                 <h3 id="clientStatus"></h3>
             </div>
         </div>
+        <a id="download" download="clientPortfolio.json" style="display:none"></a>
```
Comment (flashd2n) [inline tutorials/javascript/start/client-details/index.html:31]: to remove

**Review by flashd2n — CHANGES_REQUESTED** [review]

File: tutorials/javascript/start/client-details/index.html (hunk 3 above)
Comment (AleksTanev) [reply tutorials/javascript/start/client-details/index.html]: done!

**Review by flashd2n — APPROVED** [review]

**Pull request merged**